### DISCUSSION TITLE
Allow multiple configs if multiple filters match the same app and unload those extra configs as the filters stop matching

### DISCUSSION THREAD

**Discussion by WildGenius** [espanso/espanso#1269] [Feature requests and Ideas]
Apologies if there is already a way to accomplish what I want and I'm just not experienced enough. From what I understand, only one config may be active, and the first one that matches a filter is the one that will be used as long as that filter is matched. I have a config for my browser that uses `filter_exec` and loads a few match files via `extra_includes`. I want to create an auxiliary config for a specific browser tab that will use `filter_title` and load an additional match file via `extra_includes`. All match files are `_` prefixed, i.e., will not load by default. Currently, once the browser config is matched, I have no way to get at the tab matches - the tab config simply won't activate, as the browser config still applies. Since I don't want the browser matches to be active everywhere by making them load by default, I don't see a way to dynamically add and take away the tab matches as the tab is switched to and from. 

Essentially, I need a new filter to be able to supercede an already active one. I don't mind copying all the matches from the browser config into the tab config if multiple configs at the same time (allowing to include and exclude matches as each new one is loaded) is too much to ask.

[A seemingly similar request](https://github.com/federico-terzi/espanso/issues/320) was made during the `parent` system, which I know nothing about.

**Comment by federico-terzi** [comment]
Hey @WildGenius,

Thank you for the feedback! Interesting question, could you please try prefixing the name of the config file (the `filter_title` one) with an "a"? I'm wondering if having it come before the `filter_exec` one in alphabetical ordering could work

So if your `filter_title` config is `firefox.yml`, call it `afirefox.yml`

Let me know if that helps

Cheers :)

**Comment by WildGenius** [comment]
Shame on me for missing such a basic workaround! Indeed, after renaming, the "smaller" tab `filter_title` config is switched to correctly. Obviously, I do have to copy over the more general browser matches inside to keep them, so it technically becomes "bigger", but the result is as desired.

I think the functionality in the OP would still be useful for easier granular inclusion/exclusion of matches.

Not to make excuses, but any chance the precise rules for loading configs (is there a particular reason you suggested a letter prefix rather than a symbol?..) could be added to [the documentation's page on app-specific configs](https://espanso.org/docs/configuration/app-specific-configurations/)? I had naively assumed that the only name that mattered was `default.yml`.

Thanks for the app and for the prompt response!

**Reply by smeech** [reply]
It might be possible to use the techniques at: [Include and Exclude rules](https://espanso.org/docs/configuration/include-and-exclude/) to avoid the need to copy the browser matches.

**Reply by WildGenius** [reply]
All matches are being loaded via `extra_includes` (see OP). In this case, "copying" refers to having to have the browser matches inside both the broad browser config's and the granular tab config's `extra_includes`. In other words, it's the lines in the configs that are copied, not the trigger-replacement pairs themselves.

**Reply by smeech** [reply]
Sorry for overlooking that - you clearly know what you're doing!

**Comment by my-world-2022** [comment]
I also have this need

**Comment by smeech** [comment]
One for the documentation.
